pandas/core/groupby/groupby.py

```diff
@@ -343,14 +345,25 @@ def _group_selection_context(groupby):
     groupby._reset_group_selection()
 
 
+KeysArgType = Optional[
+    Union[
+        Hashable,
+        List[Hashable],
+        Callable[[Hashable], Hashable],
+        List[Callable[[Hashable], Hashable]],
```

Comment:
Pretty large signature and maybe worth trimming down / cleaning up (see #22278) but documenting as-is state

Reply:
yikes.  maybe a comment pointing to the relevant issue?